include/geos/io/GeoJSON.h

```diff
@@ -0,0 +1,109 @@
+/**********************************************************************
+ *
+ * GEOS - Geometry Engine Open Source
+ * http://geos.osgeo.org
+ *
+ * Copyright (C) 2011 Sandro Santilli <strk@kbt.io>
+ * Copyright (C) 2005-2006 Refractions Research Inc.
+ * Copyright (C) 2001-2002 Vivid Solutions Inc.
+ *
+ * This is free software; you can redistribute and/or modify it under
+ * the terms of the GNU Lesser General Public Licence as published
+ * by the Free Software Foundation.
+ * See the COPYING file for more information.
+ *
+ **********************************************************************/
+
+#ifndef GEOS_IO_GEOJSON_H
+#define GEOS_IO_GEOJSON_H
+
+#include <geos/export.h>
+
+#include <map>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <cctype>
+#include <geos/geom/Geometry.h>
+
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable: 4251) // warning C4251: needs to have dll-interface to be used by clients of class
+#endif
+
+namespace geos {
+namespace geom {
+class Geometry;
+}
+}
+
+namespace geos {
+namespace io {
+
+enum class GeoJSONValueType {
+    STRING, NUMBER, BOOLEAN, NULLTYPE, OBJECT, ARRAY
+};
+
+struct GeoJSONValue {
+
+    GeoJSONValueType type;
```

Comment:
Maybe a union?

Reply:
I converted this to a union and it turned out to be tricky because of std::string, std::map, and std::vector.  I